fl4health/preprocessing/autoencoders/dim_reduction.py

```diff
@@ -26,59 +28,113 @@ def __repr__(self) -> str:
         return f"{self.__class__.__name__}"
 
 
-class AeProcessor(Processing):
-    """Transformer processor to encode the data using basic autoencoder."""
-
-    def __init__(
-        self,
-        checkpointing_path: Path,
-    ) -> None:
-        super().__init__(checkpointing_path)
+class AeProcessor(AutoEncoderProcessing):
+    """
+    Transformer processor to encode the data using basic autoencoder.
+    """
 
     def __call__(self, sample: torch.Tensor) -> torch.Tensor:
         # This transformer is called for the input samples after they are transfered into torch tensors.
         embedding_vector = self.autoencoder.encode(sample.to(self.device))
         return embedding_vector.clone().detach()
 
 
-class VaeProcessor(Processing):
-    """Transformer processor to encode the data using VAE encoder."""
-
+class VaeProcessor(AutoEncoderProcessing):
     def __init__(
         self,
         checkpointing_path: Path,
-        include_variance: bool = False,
+        return_mu: bool = False,
```

Comment:
I would suggest calling this `return_mu_only`? When I first return this, I thought if it were false we wouldn't return the mean, but when its false what we really mean is that we're not _only_ returning mu.